control_plane/attachment_service/src/persistence.rs

```diff
@@ -381,16 +381,19 @@ impl Persistence {
         self.with_conn(move |conn| -> DatabaseResult<()> {
             conn.transaction(|conn| -> DatabaseResult<()> {
                 // Mark parent shards as splitting
+
+                let expect_parent_records = std::cmp::max(1, old_shard_count.0);
+
                 let updated = diesel::update(tenant_shards)
                     .filter(tenant_id.eq(split_tenant_id.to_string()))
                     .filter(shard_count.eq(old_shard_count.0 as i32))
                     .set((splitting.eq(1),))
                     .execute(conn)?;
-                if ShardCount(updated.try_into().map_err(|_| DatabaseError::Logical(format!("Overflow existing shard count {} while splitting", updated)))?) != old_shard_count {
+                if u8::try_from(updated).map_err(|_| DatabaseError::Logical(format!("Overflow existing shard count {} while splitting", updated)))? != expect_parent_records {
```

Comment:
Could just split these up, rustfmt would prefer that as well. It seems to have already given up here as well.

```suggestion
                let updated = u8::try_from(updated)
                    .map_err(|_| DatabaseError::Logical(
                        format!("Overflow existing shard count {updated} while splitting"))
                    )?;
                if updated != expect_parent_records {
```